src/main/java/mate/academy/dao/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+package mate.academy.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.sql.Types;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    private static final String CREATE_SQL = "INSERT INTO books (title, price) VALUES(?, ?)";
+    private static final String FIND_BY_ID_SQL = "SELECT * FROM books WHERE id = ?";
+    private static final String FIND_ALL_SQL = "SELECT * FROM books";
+    private static final String UPDATE_BY_ID_SQL = "UPDATE books SET title = ?, "
+            + "price = ? WHERE id = ?";
+    private static final String DELETE_BY_INDEX_SQL = "DELETE FROM books WHERE id = ?";
+
+    @Override
+    public Book create(Book book) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement statement = connection
+                        .prepareStatement(CREATE_SQL, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, book.getTitle());
+            statement.setBigDecimal(2, book.getPrice());
+            int affectedRows = statement.executeUpdate();
+            if (affectedRows < 1) {
+                throw new DataProcessingException("Expected to insert at list one row"
+                        + "but were inserted zero rows");
+            }
+            ResultSet generatedKeys = statement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                book.setId(generatedKeys.getObject(1, Long.class));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can`t create new book " + book, e);
+        }
+        return book;
+    }
+
+    @Override
+    public Optional<Book> findById(Long id) {
+        Book book = new Book();
```

Comment:
Initializing a new `Book` object here is unnecessary since it will be overwritten if a book is found. Consider initializing it only if a book is found in the database.